packages/lib/services/rest/routes/notes.ts

```diff
@@ -103,11 +103,18 @@ async function requestNoteToNote(requestNote: any) {
 		output.parent_id = folder.id;
 	}
 
+	if ('is_todo' in requestNote) {
+		output.is_todo = Database.formatValue(Database.TYPE_INT, requestNote.is_todo);
+		if (output.is_todo === 1) {
+			if ('todo_due' in requestNote) output.todo_due = Database.formatValue(Database.TYPE_INT, requestNote.todo_due);
+			if ('todo_completed' in requestNote) output.todo_completed = Database.formatValue(Database.TYPE_INT, requestNote.todo_completed);
+		}
+	}
+
 	if ('source_url' in requestNote) output.source_url = requestNote.source_url;
 	if ('author' in requestNote) output.author = requestNote.author;
 	if ('user_updated_time' in requestNote) output.user_updated_time = Database.formatValue(Database.TYPE_INT, requestNote.user_updated_time);
 	if ('user_created_time' in requestNote) output.user_created_time = Database.formatValue(Database.TYPE_INT, requestNote.user_created_time);
-	if ('is_todo' in requestNote) output.is_todo = Database.formatValue(Database.TYPE_INT, requestNote.is_todo);
 	if ('markup_language' in requestNote) output.markup_language = Database.formatValue(Database.TYPE_INT, requestNote.markup_language);
 	if ('longitude' in requestNote) output.longitude = requestNote.longitude;
 	if ('latitude' in requestNote) output.latitude = requestNote.latitude;
```

Comment:
Why don't we add support for todo_due and todo_completed here, instead of making it a special case? That would fix the original bug, wouldn't it?

Reply:
Yes, that does fix the bug.